scripts/managedb/csubs_control/csub_dups_checker.py

```diff
@@ -0,0 +1,163 @@
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+__author__ = 'fermin'
+
+import getopt
+import hashlib
+import pickle
+import getpass
+import time
+from sys import argv, exit
+from pymongo import MongoClient, DESCENDING
+from bson.objectid import ObjectId
+from subprocess import call
+
+def msg(s):
+    if verbose:
+        print s
+
+def usage():
+    print '%s --db <database> [--with-restart <status_file>] [--dry-run] [-v]' % basename
+
+def log_status_restart():
+    now = time.strftime('%d-%m-%Y %H:%M:%S')
+    with open(status_file, 'a') as file:
+        file.write('%s - contextBroker restart by %' %(now, basename))
+
+# Argument parsing
+basename = __file__
+DB = None
+with_restart = False
+dry_run = False
+verbose = False
+try:
+   opts, args = getopt.getopt(argv[1:], 'v', ['db=', 'with-restart=', 'dry-run'])
+except getopt.GetoptError as err:
+   print str(err)
+   usage()
+   exit(1)
+for o, a in opts:
+   if o == '--db':
+       DB = a
+   elif o == '--with-restart':
+       with_restart = True
+       status_file = a
+   elif o == '--dry-run':
+       dry_run = True
+   elif o == '-v':
+       verbose = True
+   else:
+       assert False, 'unhandled option'
+
+if DB == None:
+    print 'ERROR: missing db name'
+    exit(1)
+
+if with_restart and getpass.getuser() != 'root':
+    print 'ERROR: only root can launch this script using --with-restart'
+    exit(1)
+
+COL = 'csubs'
+msg('INFO: parameter DB: %s' % DB)
+msg('INFO: parameter with_restart %s' % str(with_restart))
+msg('INFO: parameter dry_run: %s' % str(dry_run))
+
+client = MongoClient('localhost', 27017)
+db = client[DB]
+
+# csub document processing is based on the model described at
+# https://github.com/telefonicaid/fiware-orion/blob/develop/doc/manuals/admin/database_model.md#csubs-collection
+
+# Firs stage: analyze csubs documents
+duplicates = {}
+
+# Note that sorting by {expiration: -1} will get subscriptions by decreasing expiration order, thus
+# in the case of deleting a duplicated document, the first one in the duplicated set (i.e. the
+# one with higher expiration time) is the one that is preserved. Using a different ordering order based
+# on _id we could change the policy to preserve the first csub created
+n = 0
```

Comment:
I have the doubt between keep the "last created one" (using creation time) or "the one that will last longer, no matter when was created" (using expiration). At the end, I chose the last criteria, which I think is better (if csubs are equal, it doesn't matter which one where created first).